modules/gbk_splitter.nf

```diff
@@ -1,30 +1,27 @@
 #!/usr/bin/env nextflow
 nextflow.enable.dsl=2
 
-process gbk_split {
-    publishDir "$launchDir/icefinder_results/gbk", pattern: '*.gbk', mode: 'copy'
-    publishDir "$launchDir/icefinder_results/", pattern: 'input.list', mode: 'copy'
-
-    memory "4 GB"
-    cpus 1
+process GBK_SPLITTER {
+    publishDir "$launchDir/$params.outdir/prediction/icefinder_results/gbk", pattern: '*.gbk', mode: 'copy'
+    publishDir "$launchDir/$params.outdir/prediction/icefinder_results/", pattern: 'input.list', mode: 'copy'
 
     input:
-      path gbk_file, name: 'contigs.gbk'
+      path gbk_file
       
     output:
       path "*.gbk"
       path "input.list" , emit: gbks
 
-    tmpDir = file("$launchDir/icefinder_results/tmp")
+    tmpDir = file("$launchDir/$params.outdir/prediction/icefinder_results/tmp")
```

Comment:
This is very unusual, and it's likely to be a problem in some cases. For example running this in AWS is likely to be a problem

Reply:
I have removed the launchDir bit from the path. I am generating the directories in this step previous to ICEfinder because if the directories don't exist in advance, the tool crashes :(